### PR TITLE
PCVDeposit V2

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

File: contracts/pcv/ERC20HoldingPCVDeposit.sol [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        if (ethBalance != 0) {
+            if (block.chainid == 1) {
+                Constants.WETH.deposit{value: ethBalance}();
+            } else if (block.chainid == 42161) {
+                Constants.ARBITRUM_WETH.deposit{value: ethBalance}();
+            } else {
+                revert("Can only wrap eth on mainnet and arbitrum");
+            }
+        }
+    }
```
Comment (k-xo) [inline contracts/pcv/ERC20HoldingPCVDeposit.sol:109]: For this we could store the WETH addresses for different chains in some sort of mapping, and pass the `chainId` instead to this function, that way if we want to deploy on say optimism we wouldn't have to redeploy this contract, wdyt?

Comment (ElliotFriedman) [reply contracts/pcv/ERC20HoldingPCVDeposit.sol:109]: we can just add the optimism param in here, that will be cheaper than doing an SLOAD to figure out what network we're on

File: contracts/pcv/compound/CompoundPCVDepositBase.sol [hunk 2]
```diff
@@ -38,15 +27,21 @@ abstract contract CompoundPCVDepositBase is PCVDeposit {
...
     {
-        require(
-            cToken.redeemUnderlying(amountUnderlying) == 0,
-            "CompoundPCVDeposit: redeem error"
-        );
-        _transferUnderlying(to, amountUnderlying);
-        emit Withdrawal(msg.sender, to, amountUnderlying);
+        _withdraw(to, amountUnderlying);
+    }
+
```
Comment (ElliotFriedman) [inline contracts/pcv/compound/CompoundPCVDepositBase.sol:35]: Add withdrawAllAvailable which pulls the lesser of the two: amount wanted to withdraw or available liquidity in compound

File: contracts/pcv/compound/CompoundPCVDepositV2.sol [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    /// @notice withdraw tokens from the PCV allocation
+    /// @param amountUnderlying of tokens withdrawn
+    /// @param to the address to send proceeds
+    function withdraw(address to, uint256 amountUnderlying)
+        external
+        override
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+        whenNotPaused
```
Comment (ElliotFriedman) [inline contracts/pcv/compound/CompoundPCVDepositV2.sol:79]: do we want this to be pauseable?

File: contracts/pcv/compound/CompoundPCVDepositV2.sol [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        uint256 amountUnderlying = balance();
+        _withdraw(to, amountUnderlying);
+    }
+
+    /// @notice withdraw all available tokens from the PCV allocation
+    /// @param to the address to send proceeds
+    function withdrawAllAvailable(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+        whenNotPaused
```
Comment (ElliotFriedman) [inline contracts/pcv/compound/CompoundPCVDepositV2.sol:100]: do we want this to be pauseable?

File: contracts/pcv/compound/CompoundPCVDepositV2.sol [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        _withdraw(to, amountUnderlying);
+    }
+
+    /// @notice withdraw all tokens from the PCV allocation
+    /// @param to the address to send proceeds
+    function withdrawAll(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+        whenNotPaused
```
Comment (ElliotFriedman) [inline contracts/pcv/compound/CompoundPCVDepositV2.sol:89]: do we want this to be pauseable?

**Pull request closed without merging**